lumicks/pylake/lowlevel.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+
+from lumicks.pylake.kymo import Kymo
+from lumicks.pylake.scan import Scan
+from lumicks.pylake.channel import Slice, Continuous, empty_slice
+from lumicks.pylake.point_scan import PointScan
+from lumicks.pylake.detail.confocal import ConfocalFile, ScanMetaData
+from lumicks.pylake.detail.imaging_mixins import _FIRST_TIMESTAMP
+
+
+def create_confocal_object(
+    name,
+    infowave,
+    json_metadata,
+    red_channel=empty_slice,
+    green_channel=empty_slice,
+    blue_channel=empty_slice,
+) -> Kymo | Scan:
```

Comment:
I wonder why the mock version of this function got fully typed arguments while this doesn't. Is this a conscious choice?

Reply:
This was intentional. We usually do it either in the docstring (for our (semi-)public stuff) or in the parameter list, but not both to not repeat ourselves.

For internal stuff like this, we usually keep the docstring very short as people working on the code will gauge what it is doing quickly, hence I figured the type hints would be preferable here. I'm not opposed to changing it though.